src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/ConnectSqlCmdCommand.cs

```diff
@@ -0,0 +1,85 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlServer.Management.Common;
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Data.SqlClient;
+
+namespace Microsoft.SqlTools.ServiceLayer.BatchParser
+{
+    public class ConnectSqlCmdCommand : SqlCmdCommand
+    {
+        internal ConnectSqlCmdCommand(string server, string username, string password) : base(LexerTokenType.Connect)
+        {
+            Server = server;
+            UserName = username;
+            Password = password;
+        }
+
+        public string Server { get; private set; }
+        public string UserName { get; private set; }
+        public string Password { get; private set; }
+
+
+        /// <summary>
+        /// attempts to establish connection with given params
+        /// </summary>
+        /// <returns>returns the connection object is successful, throws otherwise</returns>
+        public DbConnection Connect()
+        {
+            //create SqlConnectionInfo object
+            SqlConnectionInfo tempConnectionInfo = new SqlConnectionInfo();
+            if (Server != null && Server.Length > 0)
+            {
+                tempConnectionInfo.ServerName = Server;
+            }
+            if (UserName != null && UserName.Length > 0)
+            {
+                tempConnectionInfo.UseIntegratedSecurity = false;
+                tempConnectionInfo.UserName = UserName;
+                tempConnectionInfo.Password = Password;
+            }
+            else
+            {
+                tempConnectionInfo.UseIntegratedSecurity = true;
+            }
+
+            DbConnection dbConnection = AttemptToEstablishCurConnection(tempConnectionInfo);
+            return dbConnection;
+        }
+
+        /// <summary>
+        /// called when we need to establish new connection for batch executio as a
+        /// result of "connect" command processing
+        /// </summary>
+        /// <param name="ci"></param>
+        /// <returns></returns>
+        private DbConnection AttemptToEstablishCurConnection(SqlConnectionInfo ci)
+        {
+            if (ci == null || ci.ServerName == null)
+            {
+                return null;
+            }
+
+            IDbConnection conn = null;
+            try
+            {
+                string connString = ci.ConnectionString;
+                connString += ";Pooling=false"; //turn off connection pooling (this is done in other tools so following the same pattern)
```

Comment:
why not set the SqlConnectionInfo.Pooled property instead before using ci.ConnectionString out. Is there a reason why this can't be set in the calling function (i.e in the Connect() method.

Reply:
This code is copied from the current windows code (similar in SSDT/SSMS) for handling these commands and then trimmed that's why some things are the way they are, I will try to change and see the effect for this and other suggestions.